### ISSUE TITLE
Improve DPI circumvention

### ISSUE THREAD

**Issue by piousdeer** [celzero/rethink-app#386]
RethinkDNS currently has basic anti deep packet inspection functionality, same as in GreenTunnel (https://t.me/rethinkdns/22091). It does not work in Russia, censored websites are still blocked with RethinkDNS enabled.

**Related open-source anti-DPI software**

Tested and working:
[PowerTunnel (Desktop)](https://github.com/krlvm/PowerTunnel)
[PowerTunnel (Android)](https://github.com/krlvm/PowerTunnel-Android)

Untested:
[DPITunnel (Android)](https://github.com/zhenyolka/DPITunnel)
[GoodbyeDPI (Windows)](https://github.com/ValdikSS/GoodbyeDPI)
[zapret (Linux)](https://github.com/bol-van/zapret)
[GreenTunnel (Desktop)](https://github.com/SadeghHayeri/GreenTunnel)

**Comment by ignoramous** [comment]
See also: celzero/firestack#1

**Comment by rustforfuture** [comment]
Agree, i heard last year that some users had success to use these tools, although DPI tunnel (Android) don't work for me (but may be i don't know how to use it)

**Comment by rustforfuture** [comment]
But most blocking here is harder than something that can get around with Anti DPI apps and intra.

They just make normal browsing a litter better, you can't use internet as you want if you don't use vpn. (Of course for those of us that read foreign websites mostly)

For users that just use Instagram or iranian websites there isn't much problem nor in speed nor odd annoyances.

**Comment by MicroBeast** [comment]
This's more a related feature request.

Split DPI Circumvention as own toggle/option so we could use it when using Rethink app as DNS only or Firewall only.

Or should I create new Feature Request issue?

**Comment by ignoramous** [comment]
Hm... DPI circumvention without a tunnel / proxy mostly involves mucking with the outgoing packets. This isn't possible at all in DNS-only mode.

**Comment by MicroBeast** [comment]
Got it. I understand. I kind of already thought so. But I don't have much experience in Networking to confirmed it.

Thanks.

**Comment by lepz0r** [comment]
There's also DPI Circumvention tool called [geneva](https://github.com/Kkevsterrr/geneva) but it's only for Linux but it's slow as it's written in Python but it's versatile

**Comment by Pantyhose-X** [comment]
It's been three years and it hasn't been implemented 😞

**Comment by ignoramous** [comment]
> It's been three years and it hasn't been implemented 😞

It isn't like we've been idle doing nothing ;)

**Comment by Lanius-collaris** [comment]
@Zig-VS-TypeScript-VS 
On Android 9 or newer, if "Private DNS mode" is "automatic" and the DNS server supports DNS over TLS ( e.g. 2606:4700:4700::1111 ), Android will send DNS queries over TLS.

**Comment by Lanius-collaris** [comment]
Will v055o remove the limit on the remote port? Packets sent to other ports may also be censored.
https://github.com/celzero/firestack/blob/5bb403804f4e322a546eeca062401fd93b7cac8c/intra/dialers/rdial.go#L73
https://github.com/celzero/firestack/blob/5bb403804f4e322a546eeca062401fd93b7cac8c/intra/dialers/rdial.go#L90

**Comment by ignoramous** [comment]
> Will v055o remove the limit on the remote port? Packets sent to other ports may also be censored

Not planned. You see it make any material difference (for split) given TLS is mostly 443. Desync perhaps could be turned on for all ports ...
